wasm/samples/fees_executor/src/lib.rs

```diff
@@ -0,0 +1,125 @@
+//! Iroha executor with fees support.
+
+#![no_std]
+
+#[cfg(not(test))]
+extern crate panic_halt;
+
+extern crate alloc;
+
+use dlmalloc::GlobalDlmalloc;
+use fees_executor_data_model::parameters::*;
+use iroha_executor::prelude::*;
+use iroha_executor_data_model::parameter::Parameter as _;
+
+#[global_allocator]
+static ALLOC: GlobalDlmalloc = GlobalDlmalloc;
+
+#[derive(Visit, Execute, Entrypoints, Debug, Clone)]
+#[visit(custom(
+    visit_set_parameter,
+    visit_unregister_domain,
+    visit_unregister_asset_definition,
+    visit_unregister_account,
+))]
+struct Executor {
+    host: Iroha,
+    context: Context,
+    verdict: Result,
+}
+
+/// Finds currently used fees options
+fn finds_fee_options(host: &Iroha) -> FeesOptions {
+    let parameters = host.query_single(FindParameters).dbg_unwrap();
+
+    let fees_options: FeesOptions = parameters
+        .custom()
+        .get(&FeesOptions::id())
+        .unwrap()
+        .try_into()
+        .expect("INTERNAL BUG: Failed to deserialize json as `FeesOptions`");
+
+    fees_options
+}
+
+/// Errors as a result of fees options validation
+#[derive(Debug)]
+enum FeesOptionsValidationError {
+    /// Invalid Asset
+    Asset,
```

Comment:
```suggestion
    /// Could not find asset specified in the fee options: {.0}
    AssetNotFound(AssetId),
```

Maybe also derive `thiserror::Error`?

Reply:
I didn't know they started to support `no_std` env.